internal/jujucommands/bootstrap.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2025 Canonical.
+
+package jujucommands
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	jujucloud "github.com/juju/juju/cloud"
+	"github.com/juju/juju/juju/osenv"
+	"github.com/juju/juju/jujuclient"
+	_ "github.com/juju/juju/provider/lxd"
+	"github.com/juju/version/v2"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+)
+
+// BootstrapCmdParams holds the parameters to bootstrap a controller for JIMM.
+type BootstrapCmdParams struct {
+	CloudNameAndRegion   string
+	ControllerName       string
+	AgentVersion         string
+	BootstrapTimeout     int
+	LoginTokenRefreshURL string
+}
+
+func (b BootstrapCmdParams) Validate() error {
+	if b.CloudNameAndRegion == "" {
+		return errors.New("cloud [and region] name cannot be empty")
+	}
+
+	if b.ControllerName == "" {
+		return errors.New("controller name name cannot be empty")
+	}
+
+	if b.AgentVersion != "" {
+		if _, err := version.ParseBinary(b.AgentVersion); err != nil {
+			if _, err := version.Parse(b.AgentVersion); err != nil {
+				return err
+			}
+		}
+	}
+
+	if b.BootstrapTimeout < 0 {
+		return errors.New("bootstrap timeout cannot be less than or equal to 0")
+	}
+
+	if b.LoginTokenRefreshURL == "" {
+		return errors.New("login-token-refresh-url cannot be empty")
+	}
+
+	return nil
+}
+
+func (b BootstrapCmdParams) BuildBootstrapCmdStr() string {
+	var builder strings.Builder
+	builder.WriteString("bootstrap")
+
+	builder.WriteString(fmt.Sprintf(" --login-token-refresh-url=%s", b.LoginTokenRefreshURL))
```

Comment:
```suggestion
	builder.WriteString(fmt.Sprintf("bootstrap --login-token-refresh-url=%s", b.LoginTokenRefreshURL))
```
and remove L61... the `login-token-refresh-url` flag will always be there